### PR TITLE
Adds failing tests for filtering

### PR DESCRIPTION
**Pull request by tsteuwer**
This ticket adds failing tests for #62

### CODE REVIEW

**Comment by tsteuwer** [comment]
@digitalsadhu, can you CR and merge?

**Comment by digitalsadhu** [comment]
Nice, love the post contents in the tests. Have you tried with object syntax as well? Does it make any difference? 

Eg.

**Comment by digitalsadhu** [comment]
Man replying from phone sucks :;

File: test/find.test.js [hunk 1]
```diff
@@ -204,13 +204,54 @@ describe('loopback json api component find methods', function () {
...
+        .end(function (err, res) {
+          expect(err).to.equal(null);
+          expect(res.body.data.length).to.equal(1);
+          done();
+        });
+    });
+
+    it('should filter two', function (done) {
+      request(app)
+        .get('/posts?filter[title][like]=myth')
```
Comment (digitalsadhu) [inline test/find.test.js]: .get('/posts?filter[where][content][like]=myth')

**Comment by digitalsadhu** [comment]
@tsteuwer Loopback syntax for filtering and sparse fieldsets isn't quite the same as what JSONAPI expects.
See: 
- http://jsonapi.org/format/#fetching-sparse-fieldsets
- http://jsonapi.org/format/#fetching-filtering

Suggest we map correct JSONAPI syntax to loopback syntax internally but otherwise rely on the loopback features to implement this.

**Comment by digitalsadhu** [comment]
Side note, these travis timeout errors are slightly annoying!

**Comment by digitalsadhu** [comment]
@tsteuwer Ok if I merge this one and create a new issue specifying the mapping from correct JSON API filtering to loopback filtering?

**Comment by tsteuwer** [comment]
Nah I'll just do it correctly so I don't have to look for old code to remove :)

**Comment by tsteuwer** [comment]
cancel that, go ahead and merge it and create a new issue for tracking. sorry was on my phone and had so many emails from github lol

**Comment by digitalsadhu** [comment]
:+1:

**Pull request merged**